localstack/services/stepfunctions/asl/component/state/state_execution/state_map/state_map.py

```diff
@@ -84,7 +85,7 @@ def from_state_props(self, state_props: StateProps) -> None:
         self.item_reader = state_props.get(ItemReader)
         self.item_selector = state_props.get(ItemSelector)
         self.parameters = state_props.get(Parameters)
-        self.max_concurrency = state_props.get(MaxConcurrency) or MaxConcurrency()
+        self.max_concurrency_decl = state_props.get(MaxConcurrencyDecl) or MaxConcurrency()
```

Comment:
nit: `MaxConcurrency()` is kind of `DefaultMaxConcurrencyDecl`

Logically sound 👍  It just took me a couple of seconds to understand. The typing helped me out :)